profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry.module

```diff
@@ -159,13 +211,13 @@ function _tmgmt_poetry_form_tmgmt_job_submit($form, &$form_state) {
       $origin_job->save();
     }
   }
-  else {
-    if (isset($form_state['values']['languages'])) {
-      $requested_languages = $form_state['values']['languages'];
-    }
-    else {
-      $requested_languages = $origin_job->settings['languages'];
-    }
+  // Getting requested languages from the $form_state submitted values.
+  if (isset($form_state['values']['settings']['languages'])) {
+    $requested_languages = $form_state['values']['settings']['languages'];
+  }
+  // Checking if job already has been submit to translator at least once.
+  if (!is_null($origin_job->settings) && isset($origin_job->settings['languages'])) {
```

Comment:
In this case it is a bit more efficient and I don't need to add a check if $origin_job->settings['languages'] is an array.